src/Core/Grid/Data/Factory/AttachmentGridDataFactoryDecorator.php

```diff
@@ -0,0 +1,158 @@
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Data\Factory;
+
+use Doctrine\DBAL\Connection;
+use PrestaShop\PrestaShop\Core\Grid\Data\GridData;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollection;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollectionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+use PrestaShop\PrestaShop\Core\Util\File\FileSizeConverter;
+use PrestaShopBundle\Translation\TranslatorAwareTrait;
+
+/**
+ * Class AttachmentGridDataFactoryDecorator
+ */
+final class AttachmentGridDataFactoryDecorator implements GridDataFactoryInterface
+{
+    use TranslatorAwareTrait;
+
+    /**
+     * @var GridDataFactoryInterface
+     */
+    private $attachmentDoctrineGridDataFactory;
+
+    /**
+     * @var int
+     */
+    private $employeeIdLang;
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @var FileSizeConverter
+     */
+    private $fileSizeConverter;
+
+    /**
+     * @param GridDataFactoryInterface $attachmentDoctrineGridDataFactory
+     * @param int $employeeIdLang
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param FileSizeConverter $fileSizeConverter
+     */
+    public function __construct(
+        GridDataFactoryInterface $attachmentDoctrineGridDataFactory,
+        int $employeeIdLang,
+        Connection $connection,
+        string $dbPrefix,
+        FileSizeConverter $fileSizeConverter
+    ) {
+        $this->attachmentDoctrineGridDataFactory = $attachmentDoctrineGridDataFactory;
+        $this->employeeIdLang = $employeeIdLang;
+        $this->connection = $connection;
+        $this->dbPrefix = $dbPrefix;
+        $this->fileSizeConverter = $fileSizeConverter;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData(SearchCriteriaInterface $searchCriteria)
+    {
+        $attachmentData = $this->attachmentDoctrineGridDataFactory->getData($searchCriteria);
+
+        $attachmentRecords = $this->applyModifications($attachmentData->getRecords());
+
+        return new GridData(
+            $attachmentRecords,
+            $attachmentData->getRecordsTotal(),
+            $attachmentData->getQuery()
+        );
+    }
+
+    /**
+     * @param RecordCollectionInterface $attachments
+     *
+     * @return RecordCollection
+     */
+    private function applyModifications(RecordCollectionInterface $attachments)
+    {
+        $modifiedAttachments = [];
+
+        foreach ($attachments as $attachment) {
+            if ((int) $attachment['products'] > 0) {
+                $productNamesArray = $this->getProductNames($attachment['id_attachment']);
+                $productNames = isset($productNamesArray['product_names']) ? $productNamesArray['product_names'] : '';
```

Comment:
you can use null coalescing operator for cleaner conditions:
```suggestion
                $productNames = $productNamesArray['product_names'] ?? ''; 
```